clip_retrieval/clip_inference.py

```diff
@@ -84,7 +84,13 @@ def __getitem__(self, ind):
 
         if self.enable_image:
             image_file = self.image_files[key]
-            image_tensor = self.image_transform(PIL.Image.open(image_file))
+            try:
+                pil_image = Image.open(image_file)
+            except: # Corrupted image, skip.
+                tqdm.tqdm.write(str(image_file) + ' is not a valid image file')
+                tqdm.tqdm.write("Skipping.")
+                return self.__getitem__(ind+1)
```

Comment:
I'm afraid this will make the dataloader get the next indice twice. Isn't there another way to indicate the data loader that this sample isn't valid ?

Reply:
returning None and using a custom collate fn in the dataloader should do the trick (like this https://discuss.pytorch.org/t/questions-about-dataloader-and-dataset/806/4 ) ; could you do it that way ?

Reply:
I'll give it a shot 👍